dds/DCPS/XTypes/TypeObject.h

```diff
@@ -227,6 +244,16 @@ namespace XTypes {
   typedef ACE_CDR::Octet NameHash_slice;
   typedef Fake_TAO_Array_Forany_T<NameHash, NameHash_slice, NameHash_tag> NameHash_forany;
 
+  inline bool name_hash_equal(const NameHash& x, const NameHash& y)
```

Comment:
Why is a named function used here and not `operator==`?  Is it because this is a typedef so it wouldn't be a distinct type if there was another array of the same bound and element type?

Reply:
```/Users/wilsonj/OpenDDS/OpenDDS-java-master/opendds-794/dds/DCPS/XTypes/TypeObject.h:247:15: error: overloaded
      'operator==' must have at least one parameter of class or enumeration type
  inline bool operator==(const NameHash& x, const NameHash& y)
```

Reply:
Yeah, that makes sense.  Alternatives include using `IDL::DistinctType` from `Serializer.h` or `std::array` (directly or emulated for older compilers).  But I don't think it needs to change in this PR.